src/mon/MonCap.cc

```diff
@@ -398,11 +399,64 @@ bool MonCap::is_capable(
 		   << " addr " << addr
 		   << " on cap " << *this
 		   << dendl;
+
   mon_rwxa_t allow = 0;
   for (vector<MonCapGrant>::const_iterator p = grants.begin();
        p != grants.end(); ++p) {
     if (cct)
-      ldout(cct, 20) << " allow so far " << allow << ", doing grant " << *p << dendl;
+      ldout(cct, 20) << " allow so far " << allow << ", doing grant " << *p
+		     << dendl;
+
+    if (p->network.size()) {
```

Comment:
This whole check should probably be its own function; it's *way* bigger than the other stuff we do as part of ```is_capable```.